datafusion/physical-plan/src/aggregates/group_values/row.rs

```diff
@@ -16,22 +16,25 @@
 // under the License.
 
 use crate::aggregates::group_values::GroupValues;
-use crate::row_converter::CardinalityAwareRowConverter;
 use ahash::RandomState;
+use arrow::compute::cast;
 use arrow::record_batch::RecordBatch;
-use arrow::row::{Rows, SortField};
-use arrow_array::ArrayRef;
-use arrow_schema::SchemaRef;
-use datafusion_common::Result;
+use arrow::row::{RowConverter, Rows, SortField};
+use arrow_array::{Array, ArrayRef};
+use arrow_schema::{DataType, SchemaRef};
+use datafusion_common::{DataFusionError, Result};
 use datafusion_execution::memory_pool::proxy::{RawTableAllocExt, VecAllocExt};
 use datafusion_physical_expr::hash_utils::create_hashes;
 use datafusion_physical_expr::EmitTo;
 use hashbrown::raw::RawTable;
 
 /// A [`GroupValues`] making use of [`Rows`]
 pub struct GroupValuesRows {
+    /// The output schema
+    schema: SchemaRef,
+
     /// Converter for the group values
-    row_converter: CardinalityAwareRowConverter,
+    row_converter: RowConverter,
```

Comment:
this was removed as the upstream interning was removed in  https://github.com/apache/arrow-rs/issues/4811

cc @JayjeetAtGithub